### PR TITLE
fix(disagg/npu): 2MB-align KV pool and MetadataBuffers for HCCL IPC RMA

### PR DESCRIPTION
**Pull request by whybeyoung**
Squash of:
  379fc75e1 fix(disagg): 2MB-align NPU MetadataBuffers for HCCL IPC RMA
  173ac057e fix(disagg): 2MB-align NPU KV cache per-layer slices

CANN HCCL IPC RMA (rtsIpcMemGetExportKey -> halShmemCreateHandle) requires every registered (ptr, len) to start at a 2 MB page boundary, otherwise ADXL Connect fails later with the opaque status 503900 once the peer dials in.

torch_npu's caching allocator does NOT guarantee this — even on top of mooncake's AscendAllocator (aclrtMalloc HUGE_ONLY), caching can split a 2 MB-aligned raw block into misaligned sub-blocks. Production observed:

  va=0x12d6371b8000  page_size=2097152  len=2479489024  ErrorNo:503900

The fix is intentionally minimal and isolated to NPU; GPU / CPU paths are bit-exact identical to before:

* New module sglang.srt.disaggregation._npu_align with two helpers:
    - zeros(shape, dtype, device): over-allocate 2 MB and return a view starting at the next 2 MB boundary. Allocator-agnostic — works on top of the default torch_npu allocator OR mooncake AscendAllocator.
    - zeros_aligned_segments(N, shape, ...): N independent 2 MB-aligned tensors, used for per-layer KV registration. Non-NPU devices pass through to torch.zeros.

* MetadataBuffers: 6 fields (output_ids, *_logprobs_val/idx, *_top_logprobs_val/idx, hidden_states) routed through _npu_align.zeros.

* NPUMHATokenToKVPool / NPUMLATokenToKVPool: per-layer K/V (+ index_k for NSA/MLA-with-index) routed through zeros_aligned_segments. Each per-layer ptr exposed by get_contiguous_buf_infos is then 2 MB-aligned on its own underlying allocation.

* MooncakeKVManager.register_buffer_to_engine: NPU-only hard-assert that every kv/aux/state ptr is 2 MB-aligned before batch_register, pinning future regressions to the offending source list instead of surfacing later as opaque ADXL Connect 503900. GPU paths skip this entirely.

Padding overhead is < 2 MB per registered segment.

---
### CI States

Latest PR Test (Base): :x: [Run #26284751489](https://github.com/sgl-project/sglang/actions/runs/26284751489)
Latest PR Test (Extra): :x: [Run #26284751458](https://github.com/sgl-project/sglang/actions/runs/26284751458)

### CODE REVIEW

**Comment by whybeyoung** [comment]
And  i think if  https://github.com/Ascend/pytorch/issues/139 accepted , this pr 's code can beome more clean
